### PR TITLE
feat: Branch odometer reads

### PR DESCRIPTION
**Pull request by bsdingman**

### CODE REVIEW

File: OpenMTR/Code/Digital/American.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            {
+                List<Rect> filteredRect = new List<Rect>();
+                Mat roi = new Mat(meter.SourceImage.Clone(), readout);
+                ImageUtils.AdjustImageSkew(roi);
+                ImageUtils.ColorToGray(roi, roi);
+                Cv2.GaussianBlur(roi, roi, new Size(3, 3), 0);
+                Cv2.MorphologyEx(roi, roi, MorphTypes.Close, ImageUtils.GetKernel(new Size(3, 3)));
+                Cv2.Canny(roi, roi, 50, 150);
+                Cv2.FindContours(roi, out Point[][] contours, out HierarchyIndex[] hierarchy, RetrievalModes.Tree, ContourApproximationModes.ApproxSimple);
+                foreach (Point[] point in contours)
```
Comment (cclinard84) [inline OpenMTR/Code/Digital/American.cs]: I think if you extracted this for each loop into a method you will pass guidelines for Better Code Hub

File: OpenMTR/Code/Odometer.cs [hunk 2]
```diff
@@ -20,37 +20,42 @@ public class Odometer
...
-
-            ImageUtils.ColorToGray(image, image);
-            Cv2.GaussianBlur(image, image, new Size(5, 5), 0);
-            Cv2.AdaptiveThreshold(image, image, 250, AdaptiveThresholdTypes.GaussianC, ThresholdTypes.Binary, 5, -1.2);
-            Cv2.MorphologyEx(image, image, MorphTypes.Open, ImageUtils.GetKernel(new Size(3,3)));
-            Cv2.FindContours(image, out Point[][] contours, out HierarchyIndex[] hierarchy, RetrievalModes.List, ContourApproximationModes.ApproxSimple);
-
-            foreach (Point[] point in contours)
+            string digitRead = "";
+            int count = 0;
```
Comment (cclinard84) [inline OpenMTR/Code/Odometer.cs]: Is the count variable necessary to keep here? I see where it is incremented but we don't do anything with it.

File: OpenMTR/Code/Odometer.cs [hunk 3]
```diff
@@ -85,32 +90,9 @@ private static string ReadDigitFromStates(int[] segmentStates)
             return "";
         }
 
-        private static string ReadDigits(Mat image, List<Rect> digits)
+        public static List<Rect> SortDigits(List<Rect> digits)
```
Comment (cclinard84) [inline OpenMTR/Code/Odometer.cs]: Nice work on chopping this method down.

File: OpenMTR/Code/Digital/American.cs (hunk 1 above)
Comment (bsdingman) [reply OpenMTR/Code/Digital/American.cs]: I'm unsure why it failed the last one commit when the one before it passed.

File: OpenMTR/Code/Odometer.cs (hunk 2 above)
Comment (bsdingman) [reply OpenMTR/Code/Odometer.cs]: Debug code, I forgot to remove.

File: OpenMTR/Code/Digital/American.cs (hunk 1 above)
Comment (bsdingman) [reply OpenMTR/Code/Digital/American.cs]: Working on refactoring

**Comment by bsdingman** [comment]
@cclinard84 Any ideas how to refactor the method BCH doesn't like?

**Comment by cclinard84** [comment]
Create the dictionaries in separate methods

**Pull request merged**